### ISSUE TITLE
Email Service Automation

### ISSUE THREAD

**Issue by mfong4151** [Red-Moon-Software-Apprenticeship/redmoon-console#12]
This is a feature meant to be implemented after we have all of the other basic issues/subscriptions features completed. Basically we want to create a service that sends out confirmation emails on our next server.

**Comment by Clue355** [comment]
Does this feature include employers reaching out to apprentices with a template email letting them know they are interested?

**Comment by mfong4151** [comment]
> Does this feature include employers reaching out to apprentices with a template email letting them know they are interested?

Yeah but feel free to implement as many subfunctions as you feel you can handle. Don't feel like you have the obligation to tackle every use case on your own.

**Comment by Clue355** [comment]
ok so the function are:
- send confirmation emails for stripe payments
- send emails to apprentices via template

Is this all the functionalities?

**Comment by mfong4151** [comment]
> ok so the function are:
> 
>     * send confirmation emails for stripe payments
> 
>     * send emails to apprentices via template
> 
> 
> Is this all the functionalities?

Yes, I'm sure that we'd want email confirmation for other things, but these are a good start. If Stripe does not already give payment confirmation, then we would need to implement it. Otherwise we might opt in confirmation letting the company know that they've booked the apprentice.

It might be helpful to just abstract out the mailing functionality, and see if we can do things like create several email templates. I talked to Luke last Friday, we decided that we probably want this to be implemented as a serverless function to prevent the server from being blocked if some async function doesn't resolve fast enough.
